src/components/templates/PartyMap/components/Map/MapPortal.tsx

```diff
@@ -0,0 +1,145 @@
+  );
+
+  const isUnclickable =
+    portal.visibility === RoomVisibility.unclickable ||
+    portal.type === RoomType.unclickable;
+  const isCovertRoom = portal.type && COVERT_ROOM_TYPES.includes(portal.type);
+  const shouldBeClickable = !isCovertRoom && !isUnclickable;
+
+  // All the percentages are stored as 0 to 100 in the database for historical
+  // reasons. We scale them back down here.
+  const left =
+    safeZoneBounds.left + (safeZoneBounds.width * portal.x_percent) / 100;
+  const top =
+    safeZoneBounds.top + (safeZoneBounds.height * portal.y_percent) / 100;
+  const width = (safeZoneBounds.width * portal.width_percent) / 100;
+  const height = (safeZoneBounds.height * portal.height_percent) / 100;
+
+  const roomInlineStyles = useMemo(
+    () => ({
+      left: `${left}px`,
+      top: `${top}px`,
+      width: `${width}px`,
+      height: `${height}px`,
+      zIndex: portal.zIndex,
+    }),
+    [height, left, portal.zIndex, top, width]
+  );
+
+  const [enterWithSound] = useCustomSound(portal.enterSound, {
+    interrupt: true,
+    onend: enterPortal,
+  });
+
+  const selectRoomWithSound = useCallback(
+    (e: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      if (!shouldBeClickable) return;
+      analytics.trackEnterRoomEvent(portal.title, portal.template);
+      isExternalPortal(portal) ? openUrl(portal.url) : enterWithSound();
+    },
+    [analytics, enterWithSound, portal, shouldBeClickable]
+  );
+
+  const portalImageClasses = classNames(styles.PortalImage, {
+    [styles.livePortalEvent]: isEventLive(firstEvent),
+  });
+
+  const cleanPortal = omit(selectedPortal, "bounds");
+  const isCurrentRoomSelected = isEqual(cleanPortal, portal);
+
+  const handleSelectPortal = useCallback(
+    (portal: Room) => {
+      if (isCurrentRoomSelected) {
+        unselectPortal();
+        return;
+      }
+
+      selectPortal(portal);
+    },
+    [isCurrentRoomSelected, selectPortal, unselectPortal]
+  );
+
+  const externalIconClasses = classNames(styles.externalIcon, {
+    [styles.linkHidden]: !shouldBeClickable,
+  });
+
+  return (
+    <div className={styles.MapPortal} style={roomInlineStyles}>
+      <div className={styles.PortalOnMap}>
+        <div className={portalImageClasses}>
+          <img src={portal.image_url} alt={portal.title} />
+        </div>
+        <div
+          className={styles.portalInfo}
+          ref={isCurrentRoomSelected ? portalRef : null}
+        >
+          <div className={styles.PortalTitle}>
+            <span className={styles.portalName}>{portal.title}</span>
+            <RoomAttendance room={portal} />
+            <span className={externalIconClasses} onClick={selectRoomWithSound}>
+              <span />
+            </span>
+            {portal.spaceId && (
+              <span
+                className={styles.InfoButton}
+                onClick={() => handleSelectPortal(portal)}
+              >
+                <span />
+              </span>
```

Comment:
why we need this nested span? can't we add just a self-closing span?
```suggestion
              <span
                className={styles.InfoButton}
                onClick={() => handleSelectPortal(portal)}
              />
```

Reply:
[81823af](https://github.com/sparkletown/sparkle/pull/3008/commits/81823afa93b14face968a41f2014dc43a98e655b)